### ISSUE TITLE
Docker on Mac: "cannot create /dev/null: Permission denied" 

### ISSUE THREAD

**Issue by 5t33** [solo-io/packer-plugin-arm-image#134]
I get the following error:
```
==> arm-image: /usr/bin/apt-key: 57: /usr/bin/apt-key: cannot create /dev/null: Permission denied
```
After rinning:
```
docker build . -t packer-plugin-arm-image-local:04-17-22
docker run \
  --rm \
  --privileged \
  -v /dev:/dev \
  -v ${PWD}:/build:ro \
  -v ${PWD}/packer_cache:/build/packer_cache \
  -v ${PWD}/output-arm-image:/build/output-arm-image \
  -e PACKER_CACHE_DIR=/build/packer_cache \
  packer-plugin-arm-image-local:04-17-22 build samples/raspbian_golang.json
```

Unfortunately, adding:
```
--tmpfs /tmpfs:exec,dev
```
Mac version is 10.15.7, docker version is 19.03.13

does not fix it.

I can, however, start the shell in interactive mode and run:
```
rm /dev/null
mknod /dev/null c 1 3
chmod 666 /dev/null
```
which allows me to get past the error if I then run packer in the container.

**Comment by 5t33** [comment]
Looks like this might be solved in the latest version of docker for mac: https://stackoverflow.com/questions/67117037/remote-container-ubuntulatest-dev-null-permission-denied 
```
Bug fixes and minor changes Docker Desktop now ensures the permissions of /dev/null and other devices are correctly set to 0666 (rw-rw-rw-) inside --privileged containers. Fixes [docker/for-mac#5527](https://github.com/docker/for-mac/issues/5527)
```

**Comment by 5t33** [comment]
Looks like upgrading docker for mac fixed it
